plugins/callback/mail.py

```diff
@@ -100,21 +101,25 @@ def mail(self, subject='Ansible error mail', body=None):
 
         smtp = smtplib.SMTP(self.smtphost, port=self.smtpport)
 
-        content = 'From: %s\n' % self.sender
-        content += 'To: %s\n' % self.to
+        sender_address = email.utils.parseaddr(self.sender)
+        to_addresses = email.utils.getaddresses([self.to])
+        cc_addresses = email.utils.getaddresses([self.cc])
+        bcc_addresses = email.utils.getaddresses([self.bcc])
```

Comment:
I've been thinking about the splitting for some time now. I think this is more complex than I initially hoped.

First, the plugin option handling is very rudimentary. Types are not specified (since the values come from INI / env variables, they are implicitly strings), and none are marked as required even though they are, like `sender_address`.

Then, not splitting by `,`, but by using `email.utils.getaddresses`, breaks backwards compatibility. (I don't think this is a real problem, since `"Foo, Bar" <bar@foo.org>` should have resulted in errors before, so the changed behavior for such addresses shouldn't be a problem.)

But the worst thing is: splitting this as a string prevents any future change to changing the type of these options to lists. If you provide a sting to a list of strings option, Ansible will simply split by comma (which is exactly what this plugin does at the moment).

Right now there's no way to provide a proper list to this module (since INI files do not support lists). But we could add support for providing these options from variables.

Also there's a problem with the above code: if not specified, the options will be `None` and not a string. Calling `email.utils.getaddresses([None])` will result in `[('', 'None')]`, which will start sending emails to `None`.

How about we simply change the module options to `type: list` / `elements: str`, and then use these lists instead of doing the parsing 'manually'?

Reply:
> and none are marked as required even though they are, like `sender_address`.

Should the module raise an Exception if it detects that `sender_address` hasn’t been explicitly set ?
We could also use a default value if it’s unset. Either
* `Ansible <noreply@localhost>` or
* `'Ansible <%s@%s>' % (getpass.getuser(), socket.gethostname())`

WDYT ?

Reply:
Hmm, these are all breaking changes. I think it's best to require this option in the future, and for now deprecate not having it specified. Something like (in `set_options()`):
```
if self.sender is None:
    self._display.deprecated('Not specifying a sender for the mail callback is deprecated and will be disallowed', version='6.0.0', collection_name='community.general')
```